tests/foreman/ui_airgun/test_oscappolicy.py

```diff
@@ -0,0 +1,260 @@
+                         module_org, oscap_content_path):
+    """Test successfully firstly create and then deletes OSCAP Policy."""
+    name = gen_string('alpha')
+    oscap_content_title = gen_string('alpha')
+    with session:
+        session.organization.select(org_name=ANY_CONTEXT['org'])
+        session.oscapcontent.create({
+            'file_upload.title': oscap_content_title,
+            'file_upload.scap_file': oscap_content_path,
+        })
+        session.oscappolicy.create({
+            'create_policy.name': name,
+            'scap_content.scap_content_resource': oscap_content_title,
+            'schedule.period': 'Weekly',
+            'schedule.period_selection.weekday': 'Monday',
+            'locations.resources.assigned': [module_loc.name],
+            'organizations.resources.assigned': [module_org.name],
+            'host_group.resources.assigned': [module_host_group.name]
+        })
+        assert session.oscappolicy.search(name)[0]['Name'] == name
+        session.oscappolicy.delete(name)
+        assert not session.oscappolicy.search(name)
+
+
+def test_positive_update(session, module_host_group, module_loc,
+                         module_org, oscap_content_path, oscap_tailoring_path):
+    """Test successfully firstly create and then updates OSCAP Policy."""
+    name = gen_string('alpha')
+    oscap_content_title = gen_string('alpha')
+    tailoring_file_name = gen_string('alpha')
+    cron_line = '0 3 * * *'
+    with session:
+        session.organization.select(org_name=ANY_CONTEXT['org'])
+        session.oscapcontent.create({
+            'file_upload.title': oscap_content_title,
+            'file_upload.scap_file': oscap_content_path,
+        })
+        session.oscaptailoringfile.create({
+            'file_upload.name': tailoring_file_name,
+            'file_upload.scap_file': oscap_tailoring_path
+        })
+        session.oscappolicy.create({
+            'create_policy.name': name,
+            'scap_content.scap_content_resource': oscap_content_title,
+            'schedule.period': 'Weekly',
+            'schedule.period_selection.weekday': 'Monday',
+            'locations.resources.assigned': [module_loc.name],
+            'organizations.resources.assigned': [module_org.name],
+            'host_group.resources.assigned': [module_host_group.name]
+        })
+        session.oscappolicy.update(name, {
+            'scap_content.tailoring_file': tailoring_file_name,
+            'schedule.period': 'Custom',
+            'schedule.period_selection.cron_line': cron_line
+        })
+        policy_val = session.oscappolicy.read(name)
+        assert (policy_val['scap_content']
+                ['tailoring_file'] == tailoring_file_name)
+        assert policy_val['schedule']['period'] == 'Custom'
+        assert (policy_val['schedule']
+                ['period_selection']['cron_line'] == cron_line)
+
+
+@tier2
+def test_positive_check_dashboard(session, module_host_group, module_loc,
```

Comment:
we are talking about dashboard here, so you can't just ignore:
```
self.assertEqual(self.dashboard.get_total_hosts_count(), 1)
            self.assertEqual(
                self.oscappolicy.wait_until_element(
                    locators[
                        'dashboard.hcc.hosts_percentage'
                    ] % 'Not audited').text,
                '100%'
            )
```

Functionality is not ready from framework point of view, but should be ready in a few days

Reply:
I discuss with @abalakh, original test wasn't actually checking dashboard. But still I should use more from dashboard functionality. 

Here is how I change the `view/oscappolicy.py` I will push it if you like it. ( I was not sure how I have to push it if the PR is already closed) 
```
class SCAPPolicyDetailsView(BaseLoggedInView):
    title = Text('h1[text()[contains(., "Compliance policy")]]')

    @property
    def is_displayed(self):
        return self.browser.wait_for_element(
            self.title, exception=False) is not None

    @View.nested
    class HostsBreakdownStatus(View):
        ROOT = ".//li[@data-name='Status table']"
        status_list = ItemValueList()
        total_count = TotalCount()

    @View.nested
    class HostBreakdownChart(View):
        ROOT = ".//li[@data-name='Status chart']"
        hosts_breakdown = Text('//div[@id="overview"]/span[@id="pieLabel3"]')
```

I also want to use PieChart from Widgets, but xpath to chart_title_text and chart_title_value in this example is different as in widgets.

Reply:
so you need other dashboard widgets to check? Create new PR to airgun, you don't need to perform anything extra with already closed ones

Reply:
@tstrych please proceed with your approach, no need to introduce new widget or inherit from piechart as turns out the widget is only visually similar and internally it's totally different. If something needs to be done in airgun - just rise new PR with changes, there's nothing you should do with closed PRs.